tests/test_triton_server.py

```diff
@@ -189,20 +224,29 @@ def test_get_logs(self):
 
         # Create local server logs
         self.server = TritonServerFactory.create_server_local(
-            path=TRITON_LOCAL_BIN_PATH, config=server_config, gpus=['all'])
+            path=TRITON_LOCAL_BIN_PATH, config=server_config, gpus=gpus)
         self.server.start()
         self.server.stop()
         self.server_local_mock.assert_server_process_terminate_called()
         self.assertEqual(self.server.logs(), "Triton Server Test Log")
 
-    def test_cpu_stats(self):
+    def test_cpu_stats(self, gpus=None):
+        if gpus == None:
+            gpus = ["all"]  # Avoid strange default args behavior
+
+        # Device request for docker mode
+        device_requests = [0]
+        if len(gpus) == 0 or len(
+                self.gpu_device_factory_mock.get_cuda_visible_gpus()) == 0:
```

Comment:
You should just call the `GPUDeviceFactory.get_cuda_visible_gpus()` and expect it to run your mock. For example, in this test, we are not calling `TritonServerFactory` from the mock. We just call the original version and the mock is executed.

Reply:
I'm a bit skeptical on using `GPUDeviceFactory` directly. Calling it directly certainly works, but the goal of getting the `device_requests` variable is to test the `server_docker.py` actually sets the `device_requests` correctly, so it is a bit like testing itself with itself.
I refactored setting visible GPUs to `setUp`, so the `mock_gpu_device_factory.py` no longer decides visible GPUs, which will be decided by the `test_triton_server.py`. Then, this should no longer be an issue.